### PR TITLE
Solution of missing boost/test/utils/trivial_singleton.hpp - issue #50

### PR DESCRIPTION
**Pull request by sejd0n**
Solution of missing boost/test/utils/trivial_singleton.hpp from issue #50 
It create singleton.hpp in turtle/include/turtle/detail which replace trivial_singleton.hpp from boost library

### CODE REVIEW

File: include/turtle/detail/mutex.hpp [hunk 1]
```diff
@@ -24,118 +25,118 @@
...
 namespace mock
 {
-namespace detail
-{
+	namespace detail
+	{
 #ifdef MOCK_HDR_MUTEX
-    typedef std::recursive_mutex mutex;
-    typedef std::lock_guard< mutex > scoped_lock;
+		typedef std::recursive_mutex mutex;
```
Comment (mat007) [inline include/turtle/detail/mutex.hpp]: The indentation changes should probably be reverted.

File: include/turtle/detail/singleton.hpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+//
+//  File        : $RCSfile$
+//
+//  Version     : $Revision$
+//
+//  Description : simple helpers for creating cusom output manipulators
+// ***************************************************************************
+
+#ifndef MOCK_UTILS_TRIVIAL_SIGNLETON_HPP
+#define MOCK_UTILS_TRIVIAL_SIGNLETON_HPP
```
Comment (mat007) [inline include/turtle/detail/singleton.hpp]: There's a type here `SIGNLETON` instead of `SINGLETON`

File: include/turtle/detail/singleton.hpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+namespace { BOOST_JOIN( inst, _t)& inst = BOOST_JOIN( inst, _t)::instance(); }
+
+#elif defined(__APPLE_CC__) && defined(__GNUC__) && __GNUC__ < 4
+#define MOCK_SINGLETON_INST( inst ) \
+static BOOST_JOIN( inst, _t)& inst = BOOST_JOIN (inst, _t)::instance();
+
+#else
+
+#define MOCK_SINGLETON_INST( inst ) \
+namespace { BOOST_JOIN( inst, _t)& inst = BOOST_JOIN( inst, _t)::instance(); }
```
Comment (mat007) [inline include/turtle/detail/singleton.hpp]: Ah boost inspect complains about the unnamed namespace in a header here (that's the build error on the CI)…

**Comment by sejd0n** [comment]
I fix misspelling and indentation errors, but at this point solution #55 is more advance and resolve additions issues.

**Comment by mat007** [comment]
Thank you for initiating this, it's now been superseded by #62.

**Pull request closed without merging**